cpp/include/cudf/lists/drop_list_duplicates.hpp

```diff
@@ -28,35 +28,102 @@ namespace lists {
  */
 
 /**
- * @brief Create a new lists column by extracting unique entries from list elements in the given
- * lists column.
+ * @brief Create new lists columns by copying entries from the input lists columns ignoring
+ * duplicate list entries.
  *
- * Given an input lists column, the list elements in the column are copied to an output lists
- * column such that their duplicated entries are dropped out to keep only the unique ones. The
- * order of those entries within each list are not guaranteed to be preserved as in the input. In
- * the current implementation, entries in the output lists are sorted by ascending order (nulls
- * last), but this is not guaranteed in future implementation.
+ * Given a pair of keys-values lists columns. At each row, each list entry in one column maps
+ * to a list entry in the other column by linear order and vice versa thus lists at each row
+ * in both keys and values columns always have the same size. A pair of output keys and values lists
+ * columns are generated by copying entries from the input lists columns. The duplicate entries in
+ * each list of the keys column are ignored from copying to output only the unique list entries.
+ * Entries in the values column mapped to these key entries are also copied or ignored along with
+ * them.
```

Comment:
```suggestion
 * Given a pair of keys-values lists columns where both columns have the same number of rows 
 * and the size of each corresponding row is the same, remove entries from both columns where the
 * corresponding key values are the same. 
 * 
 * A pair of output keys and values lists columns are generated by copying entries from the input lists columns. 
 * The duplicate entries in each list of the keys column are ignored during copying to output only 
 * the unique list entries. Entries in the values column mapped to these key entries are also copied or 
 * ignored along with them.
```

Reply:
I'll change that a little bit, not using "remove" but "copy from" instead. We are not modifying the input.